observer.go

```diff
@@ -16,7 +16,6 @@ import (
 
 // Observer communicates with Aerospike and helps collecting metrices
 type Observer struct {
-	conn          *aero.Connection
```

Comment:
What was the problem that this solved? Connections can take a long time to establish sometimes, especially for external auth and on busy nodes.

Reply:
Sharing a connection among requests is leading to a segfault when there are concurrent scrape requests with one encountering a connection error. It would be better and not costly to handle every request on a separate connection.

Reply:
```
time="2020-09-01T00:01:28Z" level=debug msg="Refreshing node localhost:3000"
time="2020-09-01T00:01:28Z" level=debug msg="Refreshing node localhost:3000"
time="2020-09-01T00:01:28Z" level=debug msg=EOF
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x8fafbb]
goroutine 366 [running]:
github.com/aerospike/aerospike-client-go.(*Connection).updateDeadline(0xc000198640, 0xb8b840, 0xc00063e180)
        /go/pkg/mod/github.com/aerospike/aerospike-client-go@v3.0.4+incompatible/connection.go:260 +0xab
github.com/aerospike/aerospike-client-go.(*Connection).Write(0xc000198640, 0xc000420040, 0x2a, 0x32, 0x0, 0xc00041c060, 0xc000517a30)
        /go/pkg/mod/github.com/aerospike/aerospike-client-go@v3.0.4+incompatible/connection.go:158 +0x56
github.com/aerospike/aerospike-client-go.(*info).sendCommand(0xc000302020, 0xc000198640, 0x22, 0xc00041c060)
        /go/pkg/mod/github.com/aerospike/aerospike-client-go@v3.0.4+incompatible/info.go:68 +0x46
github.com/aerospike/aerospike-client-go.newInfo(0xc000198640, 0xc000a16040, 0x2, 0x2, 0x40f27d, 0xa6d2e0, 0xc00063e120)
        /go/pkg/mod/github.com/aerospike/aerospike-client-go@v3.0.4+incompatible/info.go:48 +0x1c3
github.com/aerospike/aerospike-client-go.RequestInfo(0xc000198640, 0xc000a16040, 0x2, 0x2, 0x20, 0x9ac000, 0x10)
        /go/pkg/mod/github.com/aerospike/aerospike-client-go@v3.0.4+incompatible/info.go:57 +0x4d
main.(*Observer).requestInfo(0xc000035000, 0x3, 0xc000a16040, 0x2, 0x2, 0x1, 0x1, 0x1)
        /exporter/observer.go:219 +0xbe
main.(*Observer).refresh(0xc000035000, 0xc0001aaae0, 0xc00013ade0, 0xb90560, 0xc00013ade0)
        /exporter/observer.go:251 +0x1aa
main.(*Observer).Collect(0xc000035000, 0xc0001aaae0)
        /exporter/observer.go:190 +0xc9
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
        /go/pkg/mod/github.com/prometheus/client_golang@v1.7.1/prometheus/registry.go:446 +0x164
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
        /go/pkg/mod/github.com/prometheus/client_golang@v1.7.1/prometheus/registry.go:455 +0x599
```

Reply:
To fix this issue, just add a mutex to the `Observer` struct, and then synchronize on the collect function, eg:
```go
// Collect function of Prometheus' Collector interface
func (o *Observer) Collect(ch chan<- prometheus.Metric) {
    o.lck.Lock()
    defer o.lck.Unlock()

    ...
```